tests/test_system.py

```diff
@@ -0,0 +1,216 @@
+    c.expect('Continue.*')
+    c.sendline('')
+    c.expect('Change config.*')
+    return var, c
+
+def doInitalConfig(username, password, download_folder=None, config_folder=None):
+    if not config_folder:
+        if os.name == 'nt':
+            conf_path = 'pyload'
+        else:
+            conf_path = '.pyload'
+        configdir = os.path.join(os.path.expanduser("~"), conf_path)
+        shutil.rmtree(configdir, ignore_errors=True)
+        configdirfile = os.path.join(base_path, '..', 'module', 'config', 'configdir')
+        if os.path.exists(configdirfile):
+            os.remove(configdirfile)
+    else:
+        shutil.rmtree(config_folder, ignore_errors=True)
+    var, c = setup_start(pyloadpath)
+    if config_folder:
+        c.sendline('y')
+        c.expect('Configpath.*')
+        c.sendline(config_folder)
+        c.sendline('')
+        # finished start again
+        var, c = setup_start(pyloadpath)
+    c.sendline('n')
+    c.expect('Make basic.*')
+    c.sendline('y')
+    c.expect('Username.*')
+    c.sendline(username)
+    c.expect('Password.*')
+    c.sendline(password)
+    c.expect('Password.*again.*')
+    c.sendline(password)
+    c.expect('Enable remote access.*')
+    c.sendline('y')
+    c.expect('Language.*')
+    c.sendline('en')
+    c.expect('Downloadfolder.*')
+    if download_folder:
+        c.sendline(download_folder)
+    else:
+        c.sendline('')
+    c.expect('Max.*')
+    c.sendline('')
+    c.expect('Use Reconnect.*')
+    c.sendline('')
+    if var[0] == 'OK':
+        c.expect('Do you want to configure ssl.*')
+        c.sendline('')
+    c.expect('Do you want to configure webinterface.*')
+    c.sendline('')
+    c.expect('Activate webinterface.*')
+    c.sendline('')
+    c.expect('Address.*')
+    c.sendline('')
+    c.expect('Port.*')
+    c.sendline('')
+    c.expect('Server.*')
+    c.sendline('')
+    c.expect('Template.*')
+    c.sendline('')
+    c.expect('Hit enter.*')
+    c.sendline('')
+    if os.name == 'nt':
+        c.logfile.close()
+
+
+class TestSystem:
+
+    @classmethod
+    def setupClass(cls):
+        doInitalConfig('Mäf', 'müfo')
+
+    def test_login(self):
+        c = p_spawn(PY_PATH + ' ' + pyloadpath)
+        if os.name != 'nt':
+            c.logfile = sys.stdout
+        else:
+            lgfile = open('login_logfile.txt', 'w')
+            c.logfile = lgfile
+        # Wait pyload is running
+        c.expect('ADDON ClickNLoad: Proxy listening.*')
+        # Login via API
+        r = requests.Session()
+        u = r.post(api_url % "login", data={"username": "Mäf", "password": "müfo"})
+        self.key = u.json()
+        assert self.key is not False
+        r.close()
+        # Login Via Web Interface
+        r = requests.Session()
+        u = r.post(url % "login", data={"username": "Mäf", "password": "müfo", "submit":"login", "do":"login"})
+        content = u.text
+        assert "Incorrect username/email or password." not in content
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=4846282)